src/cpp/fastdds/utils/QosConverters.hpp

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file QosConverters.h
+ */
+
+#ifndef _FASTDDS_UTILS_QOS_CONVERTERS_HPP_
+#define _FASTDDS_UTILS_QOS_CONVERTERS_HPP_
+#ifndef DOXYGEN_SHOULD_SKIP_THIS_PUBLIC
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/publisher/qos/DataWriterQos.hpp>
+#include <fastdds/dds/subscriber/qos/DataReaderQos.hpp>
+#include <fastdds/rtps/attributes/RTPSParticipantAttributes.h>
+#include <fastrtps/attributes/PublisherAttributes.h>
+#include <fastrtps/attributes/SubscriberAttributes.h>
+#include <fastrtps/attributes/TopicAttributes.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+namespace utils {
+
+using fastrtps::PublisherAttributes;
+using fastrtps::SubscriberAttributes;
+using fastrtps::TopicAttributes;
+
+/**
+ * Obtains the DataWriterQos from the PublisherAttributes provided.
+ *
+ * @param qos Pointer to the QoS to write on
```

Comment:
Maybe adding if the parameter is an input, an output, or both could be helpful. `set_qos_from_attributes` for `DomainParticipantQos` clarifies that the `qos` parameter serves as input and output because the properties are merged, and the attributes are only an input and they are not modified. A similar analysis should be done for these methods.

Reply:
Changed in 9edf728c8 and d8fa2d1fc Refs #15417: Doxygen comments.